src/components/features/Register/Register.jsx

```diff
@@ -0,0 +1,148 @@
+import axios from 'axios';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router';
+import { Link } from 'react-router-dom';
+
+import { API_ENDPOINT } from '../../../config/config.js';
+
+import './Register.scss';
+
+const Registration = () => {
+  const [userData, setUserData] = useState({
+    identifier: '',
+    password: '',
+    file: [],
+  });
+
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    // eslint-disable-next-line no-unused-vars
+    reset,
+    trigger,
+  } = useForm();
+
+  const handleSubmitRegistration = (data) => {
+    setUserData(data);
+
+    const handleRegistration = async () => {
+      try {
+        console.log(data, 'async');
+        const response = await axios.post(`${API_ENDPOINT}register`, data);
```

Comment:
Awesome, but a few notes here:

1. We want to extract the API calls to separate files (services) - we don't want to use `axios` directly in components because it creates highly coupled code that gets difficult to maintain. Similar to how we have `login` functionality in the `AuthContext`, we should have `register` there as well. It should work like this: `component` calls `auth context` which calls `auth service` which uses `http service` to make the ajax call
2. We want to create a wrapper around `axios` - let's call that service `http`, and that service should create an `axios` instance, set some defaults (like `baseURL`, so you don't have to put `${API_ENDPOINT}register` in every call), and export the `axios` instance, which all other API services will use
3. Create an `auth` service, that will have `login` and `register` methods, and that will use the `http` service to make ajax requests